Nickvision.Aura/Network/NetworkMonitor.cs

```diff
@@ -0,0 +1,141 @@
+using System;
+using System.Net.NetworkInformation;
+using System.Runtime.InteropServices;
+using System.Threading.Tasks;
+using Tmds.DBus;
+
+namespace Nickvision.Aura.Network;
+
+/// <summary>
+/// Network state monitor
+/// </summary>
+public class NetworkMonitor : IDisposable
+{
+    public event Action<NetworkState>? _stateChanged;
+    
+    /// <summary>
+    /// Occurs when network state is changed
+    /// </summary>
+    public event Action<NetworkState> StateChanged
+    {
+        add
+        {
+            _stateChanged += value;
+            Task.Run(async () => _stateChanged.Invoke(await GetStateAsync()));
+        }
+
+        remove
+        {
+            _stateChanged -= value;
+        }
+    }
+    
+    private bool _disposed;
+    private Connection? _dbusConnection;
+    private INetworkManager? _networkManager;
+    private string[]? _networkAddresses;
+    
+    /// <summary>
+    /// Construct NetworkMonitor
+    /// </summary>
+    public NetworkMonitor()
+    {
+        _disposed = false;
+        if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
+        {
+            #pragma warning disable CS4014
+            SetupNetworkManager();
+            #pragma warning restore CS4014
+        }
+        else
+        {
+            SetupPing();
```

Comment:
I'd move the code from the method directly here since it's just used here

Reply:
It's also used in SetupNetworkManager as fallback